Clarify entity services are allowed to setup from a platform's async_setup_entry (#2774)
home-assistant/developers.home-assistant: docs/core/integration-quality-scale/rules/action-setup.md

This way we can let the user know why the service action did not work, if the targeted configuration entry is not loaded.
The validation should happen inside the service action, and should raise `ServiceValidationError` if the input is invalid.

> Entity service should be registered from the platform's `async_setup_entry`, but they should not be removed when the platform is unloaded.

Reviewer: ```suggestion
Entity service actions should be registered from the platform's `async_setup_entry`, but they should not be removed when the platform is unloaded.
```

> Entity [service actions](/docs/dev_101_services#entity-service-actions) should be registered from the platform's `async_setup_entry`, but they should not be removed when the platform is unloaded.

Reviewer 2: ```suggestion
This rule is not applicable to [entity service actions](/docs/dev_101_services#entity-service-actions), which should be registered from the platform's `async_setup_entry`.
```
I think we can make it more clear that this rule is not about those. Also, the part about them being unloaded does not belong here since the rule is not about them.

Reviewer 2: > Also, entity services should not be unregistered when the platform is unloaded, why do you say that doesn't apply?

I meant that the rule doesn't apply. That requirement should be documented, but not in this rule.

The rule is **Service actions are registered in async_setup**, and explicitly states that service actions should not be registered in "async_setup_entry". This new sentence states the opposite for entity services, thus excluding them from the rule.

How does this rule apply if the sentence is excluding them?

Alternatively, the rule should be changed to cover both (and needs a new title).
